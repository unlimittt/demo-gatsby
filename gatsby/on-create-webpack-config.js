const onCreateWebpackConfig = ({ stage, actions }) => {
  if (stage.startsWith("develop")) {
    actions.setWebpackConfig({
      resolve: {
        alias: {
          "react-dom": "@hot-loader/react-dom",
        },
      },
    })
  }
}

module.exports = onCreateWebpackConfig
