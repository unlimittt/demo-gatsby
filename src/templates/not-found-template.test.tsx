import React from 'react';
import { useStaticQuery, StaticQuery } from 'gatsby';
import NotFoundTemplate from './not-found-template';
import siteMetadata from '../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../types';
import Enzyme, { shallow } from "enzyme"
import Adapter from "enzyme-adapter-react-16"

Enzyme.configure({ adapter: new Adapter() })

describe('NotFoundTemplate', () => {
  beforeEach(() => {
    (StaticQuery as jest.MockedFunction<typeof StaticQuery | any>).mockImplementationOnce(
      ({ render }: RenderCallback) => (
        render(siteMetadata)
      ),
      (useStaticQuery as jest.MockedFunction<typeof useStaticQuery | any>).mockReturnValue(siteMetadata)
    );
  });

  it('renders correctly', () => {
    const wrapper = shallow(<NotFoundTemplate />);
    expect(wrapper).toMatchSnapshot();
  });
});
