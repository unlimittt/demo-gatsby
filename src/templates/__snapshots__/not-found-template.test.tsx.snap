// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundTemplate renders correctly 1`] = `
<Layout
  description="Test subtitle"
  title="Not Found"
>
  <Sidebar />
  <Page
    title="NOT FOUND"
  >
    <p>
      You just hit a route that doesn't exist... the sadness.
    </p>
  </Page>
</Layout>
`;
